qanta/wikipedia/cached_wikipedia.py

```diff
@@ -149,7 +151,28 @@ def load_page(self, key: str):
             return CachedWikipedia.load_page(key)
         return raw
 
+    def get_wikidata(self, key: str):
+        web_page = "https://www.wikidata.org/w/api.php?action=wbsearchentities&search=" + key + "&language=en&format=json"
+        headers = {"Accept": "application/json"}
+        req = requests.get(web_page, headers=headers)
+
+        try:
+            if not req.json()['search']:
+                log.info("No results "" + search_term + " : " + search_term)
+            else:
+                id = req.json()['search'][0]['id']
+                url = req.json()['search'][0]['concepturi']
+                headers = {"Accept": "application/json"}
+                req = requests.get(url, headers=headers)
+                claims = req.json()['entities'][id]['claims']
+                return claims
+
+        except ValueError:
+            log.info("Decoding JSON has failed : " + search_term + " : " + search_term)
+            return None
+
     def __getitem__(self, key: str):
+        unformatted_key = key
```

Comment:
I don't think the unformatted key should be used here because by default the things that will be seen here will be accesses like `cw['albert_einstein']` since we normalize things according to `format_guess`. The call to that here is really a double check to make sure it really is normalized, but I expect that the key value stays the same and since `format_guess` is idemopotent the value should stay the same no matter how many times called.

I think the better solution is to take a key formatted with `format_guess` and convert it to be compatible with wikidata either via the Wikipedia URL (this does accept underscores), or converting underscores and whatnot to whitespace.

One thing to investigate here is if we can use the data dump instead of the search functionality in the API to do this matching.